### PR TITLE
OC-13440: Check if admin user has access to study or site for api extracts and has archivedDatasetPermission

### PR DESCRIPTION
**Pull request by suthirr-openclinica**

### CODE REVIEW

**Comment by suthirr-openclinica** [comment]
jenkins retest this please

File: src/main/java/org/akaza/openclinica/controller/ScheduledExtractController.java [hunk 1]
```diff
@@ -57,15 +66,23 @@
...
+            HttpServletRequest request,
+            HttpServletResponse response) throws SchedulerException {
+        utilService.setSchemaFromStudyOid(studyOid);
         UserAccountBean userAccountBean = utilService.getUserAccountFromRequest(request);
         if (!userAccountBean.isSysAdmin() && !userAccountBean.isTechAdmin()) {
             return ResponseEntity.badRequest().headers(HeaderUtil.createFailureAlert(ENTITY_NAME, ErrorConstants.ERR_NO_SUFFICIENT_PRIVILEGES,
-                    "User must be type admin.")).body(null);
+                    "Insufficient privileges.")).body(null);
+        }
+
```
Comment (svadla-oc) [inline src/main/java/org/akaza/openclinica/controller/ScheduledExtractController.java:81]: Here you should check if study oid is valid by looking up the study.

File: src/main/java/org/akaza/openclinica/service/ValidateServiceImpl.java [hunk 2]
```diff
@@ -251,6 +256,34 @@ public boolean isUserHas_DM_MON_RoleInStudy(List<StudyUserRoleBean> userRoles, S
...
     }
+
+    public boolean isUserHasAccessToStudyOrSiteForStudy(List<StudyUserRoleBean> userRoles, String studyOid){
+        Study study = studyDao.findPublicStudy(studyOid);
+        Study tempPublicStudy = null;
+        for (StudyUserRoleBean userRole : userRoles) {
+            tempPublicStudy = study;
+            if(Role.siteRoleMap.containsKey(userRole.getRole().getId()))
+                tempPublicStudy = studyDao.getPublicStudy(userRole.getStudyId());
+            if (study.getStudyId() == tempPublicStudy.getStudyId() || (tempPublicStudy.isSite() && study.getStudyId() == tempPublicStudy.getStudy().getStudyId())) {
```
Comment (svadla-oc) [inline src/main/java/org/akaza/openclinica/service/ValidateServiceImpl.java:267]: I don't see the logic for check userRole.getStudyId in the case of study-level role.

**Comment by svadla-oc** [comment]
👍

**Pull request merged**